### PR TITLE
Various Makefile bug fixes and improvements

### PR DESCRIPTION
**Pull request by rocky**
We need to run python mathics/test.py *twice*
Once with -o, and *then* once with -t. We can't combine these options.

Add top-level "clean" target.
Add .PHONY targets and do it properly
Turn a make comment into a remake-style comment.

### CODE REVIEW

**Review by rocky — COMMENTED** [review]
@mmatera  Did you delete (the buggy) pymathics/Makefile? The diff I saw looked like you just removed some spaces to setting the `RM` variable, but the file disappeared in master. If you need it deleted that's okay. 

In my rush I've been making too many errors. I need to stop working on this project for a little bit.

**Comment by mmatera** [comment]
> @mmatera Did you delete (the buggy) pymathics/Makefile? The diff I saw looked like you just removed some spaces to setting the `RM` variable, but the file disappeared in master. If you need it deleted that's okay.
> 
> In my rush I've been making too many errors. I need to stop working on this project for a little bit.

No, I didn't erase it (voluntarily, at least). I couldn't work too much today so I just was doing small tests.

File: Makefile [hunk 1]
```diff
@@ -28,7 +28,14 @@ install:
 
 check: pytest doctest
 
-# Run py.test tests. You can environment variable o for pytest options
+# FIXME More directories will be added as Makefiles get improved
+#: Remove derived files
+clean:
+	for dir in mathics/doc; do \
+	   $(MAKE) -C $$dir clean; \
```
Comment (GarkGarcia) [inline Makefile:35]: We should place `"`s around `$$dir`.

File: pymathics/Makefile [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+develop:
+	$(PIP) install -e .
+
+#: Set up to run from source
+install:
+	$(PYTHON) setup.py install
+
+#: Run tests. For now there's not much here
+check:
+	python test.py
```
Comment (GarkGarcia) [inline pymathics/Makefile:30]: We should pass `-o` to `test.py` (so that the `data` files are saved).

Comment (GarkGarcia) [reply pymathics/Makefile:30]: Never mind, we should definitively not do that.

File: Makefile (hunk 1 above)
Comment (GarkGarcia) [reply Makefile]: Fixed on https://github.com/mathics/Mathics/pull/887/commits/c7be29e0337a3a4a86d20d832f706b37c841f4ef.

**Pull request merged**